neighbor/CMakeLists.txt

```diff
@@ -37,34 +39,34 @@ set(_h_sources
     )
 ###########################
 
-# compile cuda sources
-CUDA_COMPILE(_CUDA_GENERATED_FILES ${_cu_sources} OPTIONS ${CUDA_ADDITIONAL_OPTIONS} SHARED)
-
 # setup library with cc sources
-add_library(neighbor SHARED ${_cc_sources} ${_CUDA_GENERATED_FILES})
-target_link_libraries(neighbor PRIVATE ${HOOMD_LIBRARIES})
-# if we are compiling with MPI support built in, set appropriate compiler/linker flags
-if (ENABLE_MPI)
-   if(MPI_CXX_COMPILE_FLAGS)
-       set_target_properties(neighbor PROPERTIES COMPILE_FLAGS "${MPI_CXX_COMPILE_FLAGS}")
-   endif(MPI_CXX_COMPILE_FLAGS)
-   if(MPI_CXX_LINK_FLAGS)
-       set_target_properties(neighbor PROPERTIES LINK_FLAGS "${MPI_CXX_LINK_FLAGS}")
-   endif(MPI_CXX_LINK_FLAGS)
-endif(ENABLE_MPI)
-fix_cudart_rpath(neighbor)
+pybind11_add_module(_${COMPONENT_NAME} SHARED ${_cu_sources} ${_cc_sources} ${_h_sources} NO_EXTRAS)
 
-# install the library
-if (HOOMD_INTERNAL)
-    install(TARGETS neighbor
-            LIBRARY DESTINATION ${PYTHON_MODULE_BASE_DIR})
+# alias into the HOOMD namespace so that plugins and symlinked components both work
+add_library(HOOMD::_${COMPONENT_NAME} ALIAS _${COMPONENT_NAME})
+
+target_link_libraries(_${COMPONENT_NAME} PUBLIC HOOMD::_hoomd pybind11::pybind11)
+
+# alias into the HOOMD namespace so that plugins and symlinked components both work
+add_library(HOOMD::_${COMPONENT_NAME} ALIAS _${COMPONENT_NAME})
+
+if (APPLE)
+set_target_properties(_${COMPONENT_NAME} PROPERTIES INSTALL_RPATH "@loader_path/..;@loader_path")
 else()
-    set_target_properties(neighbor PROPERTIES PUBLIC_HEADER "${_h_sources}")
-    install(TARGETS neighbor
-            LIBRARY DESTINATION lib
-            PUBLIC_HEADER DESTINATION include/neighbor)
+set_target_properties(_${COMPONENT_NAME} PROPERTIES INSTALL_RPATH "\$ORIGIN/..;\$ORIGIN")
 endif()
 
+# link the library to its dependencies
+target_link_libraries(_${COMPONENT_NAME} PUBLIC HOOMD::_hoomd)
```

Comment:
This is a duplicate of line 48. Does it need to be here?

Reply:
fixed